### PR TITLE
Add support for variables in Markdown

### PR DESCRIPTION
**Pull request by nkshah2**
## Summary of change
Adds support for using variables in markdown files using the custom `^{variableName}` syntax using a config file.

## Related issues
- #47 

## Test cases verified
- Single variable in line
- Variable in normal markdown text
- Variable in markdown links (with broken link checker enabled)
- Multiple and different variables in a single line
- Multiple of the same variable in a single line
- Variable in markdown code blocks
- Variable in markdown bullet points

## Remaining TODOs for this PR
- [x] Add code documentation for what the plugin script does
- [x] Changes to the contribution guide to explain how to use variables in markdown when making changes to documentation pages

### CODE REVIEW

File: v2/src/plugins/markdownVariables.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "exactfolderrecipename": {
```
Comment (rishabhpoddar) [inline v2/src/plugins/markdownVariables.json:2]: This is just an example right? Perhaps, instead, put a real example and modify the sign out page across all recipes to use this new plugin instead.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Comment by rishabhpoddar** [comment]
@nkshah2 
- please also update the contributing guide to write how to use this.
- Put one real example of using this (perhaps the sign out page across all recieps)

File: v2/src/plugins/markdownVariables.json (hunk 1 above)
Comment (nkshah2) [reply v2/src/plugins/markdownVariables.json:2]: Yeah this was meant as a sample, yeah Ill modify the sign out pages instead that makes more sense

**Comment by rishabhpoddar** [comment]
Please also change the change_me docs to have a page that showcases the use of this variable across:
- code blocks
- in-line code
- things like :::important sections
- lists and sub lists
- headings
- paragraphs
- links
- tabs
- instead a react component?
- code block title

**Comment by nkshah2** [comment]
Results of various element testing:

✅ - Working correctly
❌ - Not working
🚧 - Testing in progress

| Element used | Status |
| ------------- | :-----: |
| Markdown code blocks | ✅ |
| Markdown code block titles | ✅ |
| HTML code blocks | ✅ |
| Inline code | ✅ |
| Admonitions | ✅ |
| Markdown lists | ✅ |
| HTML lists | ✅ |
| Markdown Headers | ✅ |
| Normal Markdown Text | ✅ |
| React component content in markdown | ✅ |
| React component props | 🚧 |

**Comment by nkshah2** [comment]
@rishabhpoddar Ive added a page to explain some of the ways to use variables and ive also added entries in the config for that page. Should we consider that as the real example of using this?

**Comment by rishabhpoddar** [comment]
> Should we consider that as the real example of using this?

No. Please make a change to the sign out docs for each of the recipes and then if that works (along with final build and output), then this is fine.

File: v2/src/plugins/markdownVariables.json [hunk 2]
```diff
@@ -17,5 +17,29 @@
             "reactTabContent": "Tab value",
             "reactTabsDefaultValue": "orange"
         }
+    },
+    "session": {
+        "sign-out": {
```
Comment (rishabhpoddar) [inline v2/src/plugins/markdownVariables.json:22]: Remove page level.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: v2/src/plugins/markdownVariables.json (hunk 2 above)
Comment (nkshah2) [reply v2/src/plugins/markdownVariables.json]: Done

**Review by rishabhpoddar — APPROVED** [review]

**Comment by rishabhpoddar** [comment]
@nkshah2 Please change the code comments + the change_me docs to reflect the latest change in the structure where we do not need a file.

**Review by rishabhpoddar — APPROVED** [review]

**Comment by nkshah2** [comment]
@rishabhpoddar Done

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**